src/main/java/org/reaktivity/specification/socks/internal/SocksFunctions.java

```diff
@@ -21,41 +21,159 @@
 import org.kaazing.k3po.lang.el.spi.FunctionMapperSpi;
 import org.reaktivity.specification.socks.internal.types.control.SocksRouteExFW;
 import org.reaktivity.specification.socks.internal.types.stream.SocksBeginExFW;
-
+import org.reaktivity.specification.socks.internal.types.stream.SocksEndExFW;
+import org.reaktivity.specification.socks.internal.types.stream.SocksAbortExFW;
 public final class SocksFunctions
 {
     private static final int MAX_BUFFER_SIZE = 1024 * 8;
 
     @Function
-    public static byte[] routeEx(
-        String address,
-        int port)
+    public static SocksRouteExBuilder routeEx()
+    {
+        return new SocksRouteExBuilder();
+    }
+
+    @Function
+    public static SocksBeginExBuilder beginEx()
     {
-        final MutableDirectBuffer writeBuffer = new UnsafeBuffer(new byte[MAX_BUFFER_SIZE]);
-        final SocksRouteExFW routeEx = new SocksRouteExFW.Builder()
-                                                         .wrap(writeBuffer, 0, writeBuffer.capacity())
-                                                         .address(address)
-                                                         .port(port)
-                                                         .build();
-        final byte[] bytes = new byte[routeEx.sizeof()];
-        routeEx.buffer().getBytes(0, bytes);
-        return bytes;
+        return new SocksBeginExBuilder();
     }
 
     @Function
-    public static byte[] beginEx(
-        String address,
-        int port)
+    public static SocksEndExBuilder endEx()
+    {
+        return new SocksEndExBuilder();
+    }
+    @Function
+    public static SocksAbortExBuilder abortEx()
+    {
+        return new SocksAbortExBuilder();
+    }
```

Comment:
`SOCKS` has no protocol frames after the initial handshake, so by the time you start sending `DATA`, `END` or `ABORT` frames, there is no more `SOCKS` metadata to pass along in `SocksDataEx`, `SocksEndEx`, or `SocksAbortEx`.

Therefore, since these functions are not used by the scripts, please remove them from the code.